### PR TITLE
webpack -> esbuild

### LINKED ISSUE #15: Swap webpack for esbuild

### PR DESCRIPTION
**Pull request by kenichi**
closes #15 

n.b. targets `remove_vapor2` branch for now, will edit/rebase when #21 merges

* removes old package.json etc, replaces with only needed deps
* uses esbuild to watch for js changes, dart-sass for scss changes
* moves assets/static/* to priv/static/* and unignores (ignores priv/static/assets now)
* updates js/ and css/ static paths to assets/ and adjust test
* swaps `@import` for `@use` for including phoenix.css

### CODE REVIEW

File: assets/package.json [hunk 1]
```diff
@@ -17,24 +9,9 @@
     "phoenix": "file:../deps/phoenix",
     "phoenix_html": "file:../deps/phoenix_html",
     "phoenix_live_view": "file:../deps/phoenix_live_view",
+    "sass": "^1.47.0",
     "shim-keyboard-event-key": "^1.0.3",
     "topbar": "^0.1.4",
     "url-search-params-polyfill": "^8.1.0"
-  },
```
Comment (patmaddox) [inline assets/package.json:23]: woooo look at all those dev dependencies disappeared

File: lib/epi_contacts_web/templates/layout/root.html.heex [hunk 2]
```diff
@@ -6,14 +6,14 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0"/>
     <%= csrf_meta_tag() %>
     <%= live_title_tag assigns[:page_title] || "Share My Contacts" %>
-    <link nonce="<%= @conn.private[:nonce] %>" rel="stylesheet" href="<%= Routes.static_path(@conn, "/css/app.css") %>"/>
-    <script nonce="<%= @conn.private[:nonce] %>" defer type="text/javascript" src="<%= Routes.static_path(@conn, "/js/app.js") %>"></script>
+    <link phx-track-static nonce={ @conn.private[:nonce] }  rel="stylesheet" href={ Routes.static_path(@conn, "/assets/app.css") }/>
```
Comment (patmaddox) [inline lib/epi_contacts_web/templates/layout/root.html.heex:9]: interesting, what's this `phx-track-static` property?

**Review by patmaddox — APPROVED** [review]
Nice! Just a heads up that building the staging image _might_ fail - if I recall correctly with CommCareSync, it was because there's a `.dockerignore` file that ignored assets, so they weren't being copied over correctly. So if this fails to build an image after being merged to main, there's just a small change needed in `ratio-ops` to have it correctly copy the assets dir.

File: lib/epi_contacts_web/templates/layout/root.html.heex (hunk 2 above)
Comment (kenichi) [reply lib/epi_contacts_web/templates/layout/root.html.heex:9]: it should let phoenix detect/send changes to those files

> Optionally, you can add a `phx-track-static` to all script and link elements in your layout that use src and href. This way you can detect when new assets have been deployed by calling static_changed?.

https://hexdocs.pm/phoenix_live_view/installation.html#layouts
https://hexdocs.pm/phoenix_live_view/Phoenix.LiveView.html#static_changed?/1

File: lib/epi_contacts_web/templates/questionnaire/test_results.html.heex [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <p><%= gettext("Dear %{name},", name: full_name(@patient_case)) %></p>
 <p><%= gettext("As discussed during our call...")%> </p>
-<p><%= gettext("If you've experienced any COVID-19 symptoms...") %>
+<p><%= gettext("If you've experienced any COVID-19 symptoms...") %></p>
```
Comment (jc00ke) [inline lib/epi_contacts_web/templates/questionnaire/test_results.html.heex:3]: :pray:

**Review by jc00ke — APPROVED** [review]

**Pull request merged**